xdsl/dialects/llvm.py

```diff
@@ -46,6 +48,123 @@ def parse_parameters(parser: BaseParser) -> list[Attribute]:
         return [StringAttr.from_str(""), ArrayAttr.from_list(params)]
 
 
+@irdl_attr_definition
+class LLVMPointerType(ParametrizedAttribute, MLIRType):
+    name = "llvm.ptr"
+
+    type: ParameterDef[Attribute | NoneAttr]
+    addr_space: ParameterDef[IntAttr | NoneAttr]
+
+    def print_parameters(self, printer: Printer) -> None:
+        if isinstance(self.type, NoneAttr):
+            return
+
+        printer.print_string("<")
+        printer.print_attribute(self.type)
+        if not isinstance(self.addr_space, NoneAttr):
+            printer.print_string(", ")
+            printer.print_attribute(self.addr_space)
+
+        printer.print_string(">")
+
+    @classmethod
+    def untyped(cls):
```

Comment:
Why is this a class method? I would just keep this as a static method unless we actually use it with another class.